### DISCUSSION TITLE
UnitEnum support

### DISCUSSION THREAD

**Discussion by paaacman** [doctrine/doctrine2#10515] [Support Questions] [answered]
Hello,

Is there a reason why `UnitEnum` are not supported, but only `BackedEnum`?

`ReflectionEnumProperty` is using `$enum->value` to get the value, but it could check if the enum is a `UnitEnum` and use $enum->name instead.

I only found it mentioned here https://github.com/doctrine/orm/pull/9304#issuecomment-1012411532

**Comment by derrabus** [comment, marked as answer]
A backed enum is backed by a value for serialization purposes. If the author of an enum does not want to give us a backing value, we don't want to guess.

**Reply by paaacman** [reply]
In the concept of PHP Enum, the "value" of an UnitEnum is it's name. This way, we don't have to maintain name and value when they are identical.
I understand that this is decided like this and respect your choice, I asked because I didn't find any discussion about this.

Thanks.

**Reply by derrabus** [reply]
> In the concept of PHP Enum, the "value" of an UnitEnum is it's name.

No, that's not covered by the "concept of PHP Enum", it's just an assumption of yours. To quote from the [enum RFC](https://wiki.php.net/rfc/enumerations):

> By default, Enumerated Cases have no scalar equivalent. They are simply singleton objects.

But that "scalar equivalent" is what we need to build a generic solution for enums. And the RFC explicitly defines backed enums for that purpose.

**Reply by paaacman** [reply]
To quote from the [enum RFC](https://wiki.php.net/rfc/enumerations):
> This RFC introduces Enumerations to PHP. The scope of this RFC is limited to “unit enumerations,” that is, enumerations **that are themselves a value**.

That's why I say:
>  the "value" of an UnitEnum is it's name.

I now understand why you use Backed Enum, the need of a "scalar equivalent" and that Backed Enum are designed for that purpose, thanks.

**Reply by derrabus** [reply]
> > enumerations **that are themselves a value**.

Right, the enum cases themselves are the values, not the names of the enum cases.
